Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/ExchangeManager.java

```diff
@@ -189,9 +189,11 @@ protected UDDI_SPEC_VERSION getApiSpecEnum(String version) {
         String exName = StringUtils.isNotEmpty(exchangeName) ? exchangeName : getDefaultExchange();
 
         Map<String, OrganizationType> orgMap = exCache.get(exName);
-        for (Entry<String, OrganizationType> hcidKey : orgMap.entrySet()) {
-            OrganizationType org = orgMap.get(hcidKey.getKey());
-            orgList.add(syncWithOverrides(org, hcidKey.getKey()));
+        if (null != orgMap) {
+            for (Entry<String, OrganizationType> hcidKey : orgMap.entrySet()) {
+                OrganizationType org = orgMap.get(hcidKey.getKey());
```

Comment:
you can get this value from the Entry instead. No need to traverse the map again.